src/main/java/com/nixsolutions/repository/BusRepository.java

```diff
@@ -18,49 +19,69 @@ public BusRepository() {
         buses = new LinkedList<>();
     }
 
+    //filter
     @Override
-    public Bus getById(String id) {
+    public Optional<Bus> findById(String id) {
+        Optional.ofNullable(id).orElseThrow(()-> new IllegalArgumentException("id is null"));
         return buses.stream()
                 .filter(bus -> bus.getId().equals(id))
-                .findFirst()
-                .orElse(null);
+                .findFirst();
     }
 
     @Override
     public List<Bus> getAll() {
         return buses;
     }
 
+    //ifPresent
+    //orElseThrow
     @Override
-    public boolean create(Bus bus) {
-        return buses.add(bus);
+    public boolean save(Bus bus) {
+        Optional<Bus> optionalBus = Optional.ofNullable(bus);
+        if (findById(optionalBus.orElseThrow().getId()).isEmpty()){
+            optionalBus.filter(vehicle-> findById(vehicle.getId()).isEmpty())
+                    .ifPresent(buses::add);
+            return true;
+        }
+        return false;
```

Comment:
Как-то сложно, несколько обращение поиска одного элемента, я бы делал как-то так
```suggestion
        Optional.ofNullable(bus)
                .map(vehicle -> findById(bus.getId()))
                .filter(Optional::isEmpty)
                .ifPresent(o -> buses.add(bus));
```